packages/concerto-vocabulary/lib/vocabulary.js

```diff
@@ -145,15 +145,42 @@ class Vocabulary {
      */
     validate(modelFile) {
         const getOwnProperties = (d) => {
-            // ensures we have a valid return, even for scalars
-            return d.getOwnProperties?.() ? d.getOwnProperties?.() : [];
+            // ensures we have a valid return, even for scalars and map-declarations
+            return d.isMapDeclaration() ? [d.getKey(), d.getValue()] :  d.getOwnProperties?.() ? d.getOwnProperties?.() : [];
+        };
+
+        const getPropertyName = (p) => {
+            if(p.isKey?.()) {
+                return 'KEY';
+            } else if(p.isValue?.()) {
+                return 'VALUE';
+            } else {
+                return p.getName();
+            }
+        };
+
+        const checkProperties = (k, p) => {
```

Comment:
What is this checking?

Reply:
Checks whether if property given in .voc yaml is present inside the model file.

Reply:
Maybe better to rename it to `checkPropertyExists`, or `doesPropertyExist`?

Reply:
Changed to `checkPropertyExists`. Resolved